workflows/offline_ml_diags/offline_ml_diags/_helpers.py

```diff
@@ -46,6 +48,23 @@
 ScalarMetrics = Dict[str, Mapping[str, float]]
 
 
+def get_variable_indices(
+    data: xr.Dataset, variables: Sequence[str]
+) -> Mapping[str, Tuple[int, int]]:
+    if "time" in data.dims:
+        data = data.isel(time=0).squeeze(drop=True)
+    stacked = data.stack(sample=["tile", "x", "y"])
+    variable_dims = count_features_2d(
+        variables, stacked.transpose("sample", ...), "sample"
+    )
```

Comment:
Instead of using the previously private function, maybe use a Packer and then infer feature indices from there?  Otherwise, I'd suggest writing the logic here to detect features.  I just worry about it always being specific to the packer without intentionally and clearly linking to a packer here.  Also, I don't know if this overlaps with any of the fv3fit changes Jeremy is planning on making.

Reply:
That make sense, I'll write a copy of the logic in this module as opposed to importing the packer.